File: typescript/armstrong-numbers/armstrong-numbers.ts
export default class ArmstrongNumbers {
    static isArmstrongNumber(num: number): boolean {
        let str: string = String(num);
        let exp: number = str.length;
        let sum: number = Array.from(str)
            .map(Number)
            .reduce((acc, n) => acc + Math.pow(n, exp), 0);
        return sum == num;
    }
}

File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <math.h>

bool isArmstrongNumber(const int number) {
  char buffer[32];
  int sum = 0;
  int length = sprintf(buffer, "%d", number);

  for (int i = 0; i < length; i++) {
    sum += pow(buffer[i] - '0', length);
  }

  return number == sum;
}
